### ISSUE TITLE
🎛️ Migrate site switcher to use Stimulus approach `AutoFieldController`

### ISSUE THREAD

**Issue by lb-** [wagtail/wagtail#10035]
> ℹ️ **Part of the [Stimulus  🎛️ RFC 78](https://github.com/wagtail/rfcs/pull/78)**

### Is your proposal related to a problem?

There is a custom JavaScript implementation to add behaviour to select drop-down that will update the location (URL) when changed.

### Describe the solution you'd like

* Ensure that the 'current' option in the select drop-down for the site switcher is disabled, so that selecting it will not do anything. See wagtail/contrib/settings/forms.py (the goal here is to avoid JS logic to determine when to NOT trigger a change in the URL).
* Update the implementation of `client/src/controllers/AutoFieldController.ts` to allow for a action (event) params value that allows a custom `action` param to be added to the form before submit. See https://stimulus.hotwired.dev/reference/actions#event-objects & https://developer.mozilla.org/en-US/docs/Web/HTML/Element/form#attr-action
  * Note: This may not be the best option - happy to take suggestions for refinements here
  * Essentially we want to use the form `get` submit to do the location change, instead of updating the `window.location.url`
  * However, we need to ensure the right page is loaded, hence we need to revise `action` dynamically when the user selects the option
* Remove the jQuery implementation completely wagtail/contrib/settings/static_src/wagtailsettings/js/site-switcher.js
* Update the select field to have the suitable data attributes wagtail/contrib/settings/forms.py & the select options to also have the correct data attributes

#### Additional notes

* Remember that Site Settings is not available in the bakery demo by default, you will need to add this locally to validate the behaviour https://docs.wagtail.org/en/stable/reference/contrib/settings.html

**Comment by lb-** [comment]
Flagging as a good first issue, however, whoever does this will need to read up on Stimulus to get an understanding of the code.

@Lovelyfin00  - do you mind reviewing this issue and seeing if we should add any links/notes to make it easier to pick up. This may be a great chance for you to assist someone contributing if anyone picks this up. This will also be a good chance for you to have a think about how we document Stimulus for developers that make contributions to Wagtail.

**Comment by lb-** [comment]
Raised https://github.com/wagtail/bakerydemo/issues/395 for bakerydemo, it would be great to have a SiteSetting as part of the default demo data.

**Comment by Lovelyfin00** [comment]
That was very fast😅😅
Is there still a need for me to take a look at it?

**Comment by lb-** [comment]
@Lovelyfin00  yes, please read through the issue, I think we should probably update it with a bit more detail.
From the looks of https://github.com/wagtail/wagtail/pull/10039 (first glance) I do not think it will work, but maybe the issue description is a bit unclear.

**Comment by Lovelyfin00** [comment]
Ok great 👍

**Comment by lb-** [comment]
Note: It turns out that `change` is not triggered when the existing selected element is selected (at least in Firefox). If that can be validated in multiple browsers (or maybe we find some confirmation online) then we do not need to worry about the code handling the 'current' value NOT triggering a URL update.

FYI @Aadijindal07 

https://developer.mozilla.org/en-US/docs/Web/API/HTMLSelectElement
https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/change_event

**Comment by laymonage** [comment]
I'm pretty sure that's the case in other browsers as well; the `change` event is only fired if the value of the `<select>` input actually changed, not when the dropdown is closed (that's what `blur` is for).

**Comment by lb-** [comment]
Yeah - makes sense - however, the existing logic has a check for when the value selected is the same as the current URL. Maybe jQuery change works differently?

Or maybe it was an IE11 work around.

**Comment by lb-** [comment]
Now that #10098 is in - I have renamed and refined this issue description to reflect `w-submit` / `SubmitController`.
